single-param-eval/prelim-single-param-eval.R

```diff
@@ -0,0 +1,98 @@
+#              data contained in "stats_ ... .csv" files. It aims to identify an
+#              optimal parameter for two tree categories (i.e. 10+, overstory and
+#              10+/20+, all trees) based on the given parameter range.
+
+library(tidyverse)
+
+#### CONSTANTS ####
+STATS_DIR = "..." # path to folder containing match_stats.csv files
+OUT_DIR = "..." # path to save plot png
+
+PLOT_FILE_NAME = "...-plot" # file name to assign to plot png
+SINGLE_PARAM = "..." # the parameter of interest
+
+MIN_HEIGHT_COR = 0.85 # minimum acceptable height correlation
+MAX_SEN_PRE_DIFF = 0.5 # maximum acceptable difference between sensitivity and precision
+
+
+#### DATA LOADING ####
+if(!dir.exists(OUT_DIR)) {
+  dir.create(OUT_DIR, recursive = TRUE)
+}
+
+# load stats CSV files and paths
+stats_files = list.files(STATS_DIR)
+stats_paths = file.path(STATS_DIR, stats_files)
+
+# read and clean CSV files
+stats = stats_paths %>%
+  # read stats CSV files into one dataframe
+  map_dfr(read_csv) %>%
+  # calculate the absolute difference between sensitivity and precision
+  mutate(sen_pre_difference = abs(sensitivity - precision)) %>%
+  # remove unneeded stats
+  select(predicted_tree_dataset_name, canopy_position, height_cat, f_score, height_cor,
+         sen_pre_difference)
+
+
+#### DATA PROCESSING ####
+# calculate mean stats of all tree categories (10+/20+, all/overstory)
+all_cat_stats = stats %>%
+  group_by(predicted_tree_dataset_name) %>%
+  summarise(across(where(is.numeric), mean, .names = "mean_{.col}")) %>%
+  ungroup()
+
+# combine and finish cleaning stats dataframe
+stats = stats %>%
+  # filter stats to contain only 10+, overstory tree category
+  filter(canopy_position == "overstory" & height_cat == "10+") %>%
+  # merge stats (10+, overstory) and all_cat_stats (10+/20+, all/overstory)
+  merge(all_cat_stats, by = "predicted_tree_dataset_name") %>%
+  # extract the single parameter name from file name
+  mutate({{SINGLE_PARAM}} := as.numeric(str_extract(predicted_tree_dataset_name,
+                                                    "\\d*\\.?\\d+$"))) %>% # change as needed
+  # remove canopy_position and height_cat columns
+  select(-canopy_position, -height_cat, -predicted_tree_dataset_name)
+
+# verify height correlations and sensitivity/precision differences
+height_cor_are_passing = all(stats$height_cor > MIN_HEIGHT_COR) &
+                         all(stats$mean_height_cor > MIN_HEIGHT_COR)
+diff_are_passing = all(stats$sen_pre_difference < MAX_SEN_PRE_DIFF) &
+                   all(stats$sen_pre_difference < MAX_SEN_PRE_DIFF)
+
+# print out status of height correlations and sensitivity/precision differences
+height_cor_msg = paste("Height Correlations are above", MIN_HEIGHT_COR, ":",
+                       height_cor_are_passing)
+diff_msg = paste("Sensitivity - Precision Differences are below", MAX_SEN_PRE_DIFF,
+                 ":", diff_are_passing)
+cat(height_cor_msg, "\n", diff_msg)
```

Comment:
What if they're not passing for some candidate values of the parameter, but passing for others? This seems very possible, and we'd want to keep the passing ones. Right now, it looks like the test only affects what is printed to screen, so it's not a big deal, but we should keep in mind for the future that we should not necessarily drop all the parameter values if we get a fail.